client/dev/navbar.jsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+
+class NavLink extends React.Component{
```

Comment:
This does not need to be a full-blown Component. Instead, it can be a stateless functional component. That looks like this:
```javascript

const NavLink = props => {
  return (<a href={props.url}>{props.linktext}</a>);
};
```
The reason you can do this is because this component has no state of its own. The reason you would want to do this is because when React knows it does not have to manage state, it can run faster.

In fact, none of the components you have here have an internal state; they all have props that are passed down in read-only format.